pkg/collector/container.go

```diff
@@ -57,23 +57,30 @@ func Container(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelem
 	if argsMap == nil {
 		argsMap = map[string]string{}
 	}
+	// defines the output (sorted) array
+	var args []string
+
 	if addConfig {
+		// if key exists then delete key so it is excluded from the iteration after this block
 		if _, exists := argsMap["config"]; exists {
 			logger.Info("the 'config' flag isn't allowed and is being ignored")
+			delete(argsMap, "config")
 		}
-		// this effectively overrides any 'config' entry that might exist in the CR
-		argsMap["config"] = fmt.Sprintf("/conf/%s", cfg.CollectorConfigMapEntry())
+		args = append(args, fmt.Sprintf("--config=/conf/%s", cfg.CollectorConfigMapEntry()))
 		volumeMounts = append(volumeMounts,
 			corev1.VolumeMount{
 				Name:      naming.ConfigMapVolume(),
 				MountPath: "/conf",
 			})
 	}
 
-	var args []string
+	var sortedArgs []string
 	for k, v := range argsMap {
-		args = append(args, fmt.Sprintf("--%s=%s", k, v))
+		sortedArgs = append(sortedArgs, fmt.Sprintf("--%s=%s", k, v))
 	}
+	// ensure that the args are ordered which iterating over a map does not guarantee and append to output args
+	sort.Strings(sortedArgs)
+	args = append(args, sortedArgs...)
```

Comment:
Why does a new array need to be introduced here? Can you not just sort the args array above?

Reply:
We absolutely can, and I guess it depends on your intended behaviour where `config` should be the first item or not.  The simplest solution can simply be a one-liner of `sort.Strings(args)` after iterating `argsMap`, which was the first thing I suggested in the issue. In that case tho, we should definitely put more info into the tests on why the ordering (in the test) is expected; maybe I should provide more info anyway 🤔

Reply:
yeah more info would be great – i don't _think_ the ordering matters here. @Aneurysm9 may have more info though.

Reply:
I'm not sure that args being a map in the spec is correct at all.  Arguments such as `--config` and `--feature-gates` can be specified multiple times and the order (at least for `--config`) can be important as it will govern the order in which configuration sources are applied to build a final configuration.  Instead we should probably be taking a list of arguments directly and using it as-is.

Reply:
Thanks a lot for that information. Although I fully support changing `args` from a map to an array, that is going to be a breaking change and most certainly beyond this scope of the problem here.  Seems like this has already been discussed before inhttps://github.com/open-telemetry/opentelemetry-operator/issues/1089 but slightly deviates from the thought process of providing an array of `--config=<file from configmap>` into `args`.   

I will update the PR with comments and my thought process in the next couple of days and maybe a clearer picture will emerge.

Reply:
I think the approach Juraci recommended [here](https://github.com/open-telemetry/opentelemetry-operator/issues/1089#issuecomment-1257956898) is probably the way to go.  Rather than changing `args` from a map to a list we add a new field for providing a list of configuration URIs.  The feature gates flag is order independent and can be provided as a comma-separated list, so I'm less worried about that.